storage/range.go

```diff
@@ -1578,14 +1664,13 @@ func (r *Range) mergeTrigger(batch engine.Engine, merge *proto.MergeTrigger) err
 	}
 	r.stats.SetMVCCStats(batch, ms)
 
-	subsumedRng, err := r.rm.MergeRange(r, merge.UpdatedDesc.EndKey, merge.SubsumedRaftID)
-	if err == nil {
-		// Merge the timestamp caches from both ranges.
-		r.Lock()
-		subsumedRng.tsCache.MergeInto(r.tsCache, false /* clear */)
-		r.Unlock()
-	}
-	return err
+	// Clear the timestamp cache. In the case that this replica and the
+	// subsumed replica each held their respective leader leases, we
```

Comment:
I don't believe we need leader leases, as discussed, but there is definitely work to do with merge safety.